libraries/APM_Control/AP_SteerController.cpp

```diff
@@ -156,11 +184,21 @@ int32_t AP_SteerController::get_steering_out_rate(float desired_rate)
     _pid_info.P = (ToRad(desired_rate) * kp_ff) * scaler;
     _pid_info.FF = (ToRad(desired_rate) * k_ff) * scaler;
 	
-	// Calculate the demanded control surface deflection
-	_last_out = _pid_info.D + _pid_info.FF + _pid_info.P + _pid_info.I;
+    // Calculate the demanded control surface deflection
+    _last_out = _pid_info.D + _pid_info.FF + _pid_info.P + _pid_info.I;
 	
-	// Convert to centi-degrees and constrain
-	return constrain_float(_last_out * 100, -4500, 4500);
+    float derate_constraint = 4500;
+
+    // Calculate required constrain based on speed
+    if (speed > _deratespeed) {
```

Comment:
at _deratespeed = 0 , derating is always applied , this is in contradiction with "If 0 then no derating is used."